### PR TITLE
[CAP:CAP-251] Wave 5 F2 — manual CSV bank reconciliation (#965)

### PR DESCRIPTION
**Pull request by louisburroughs**
## Capability & Traceability
- Capability: `cap:CAP-251`
- Parent STORY (durion): louisburroughs/durion#251
- Child issue: louisburroughs/durion-positivity-backend#965 (F2 — manual CSV bank reconciliation)
- Domain: `domain:accounting`

## Contract References (REQUIRED for backend PRs touching API/event behavior)
- Contract guide entry (durion): `domains/accounting/.business-rules/BACKEND_CONTRACT_GUIDE.md` → bank reconciliation endpoints (`/v1/accounting/reconciliations/*`)
- Plan: durion `domains/accounting/plan-odoo-parity-pos-accounting.md` §7 story F2, decisions D-5/D-6.

## Contract Chain (Controller added)
- [x] OpenAPI annotations on the controller (`BankReconciliationController`)
- [x] `openapi.yaml` (+ `pos-api-gateway/docs/openapi-aggregate.yaml`) regenerated
- [ ] Angular SDK regenerated — follows in the F2 frontend work (this PR is backend-only)

## Scope
- **What:** Wave 5 story F2 — manual, CSV-only bank reconciliation backend.
  - Endpoint family under `/v1/accounting/reconciliations`: `import`, `GET`/`GET {id}`, `match`, `unmatch`, `adjustment-types`, `adjustments`, `finalize`, `report`, `audit` (10 operations).
  - Signed-amount matching, 1-to-1 and N-to-1 within ±0.01; finalize balance gate (statement vs GL ending balance within ±0.01); adjustments post **real balanced JEs** via posting categories and respect accounting-period locks (B2).
  - D-6 adjustment-type enum (`BANK_FEE, NSF_FEE, INTEREST_EARNED, FLOAT_ADJUSTMENT, OTHER`) served by `GET /adjustment-types` (frontend never hardcodes it).
  - New relational model + Flyway **V18** (`bank_reconciliation` + `_line` + `_adjustment` + `_gl_match`), replacing the dead/malformed `Reconciliation` scaffold. `ReconciliationRecord` was **kept** (it is live — used by `PaymentOutcomeProcessingServiceImpl`); only the genuinely-dead `Reconciliation` + its repository were removed.
  - Dedicated ADR-0017 error codes (`ACCOUNT_NOT_RECONCILABLE`, `RECONCILIATION_NOT_FOUND`, `RECONCILIATION_ALREADY_FINALIZED`, `MATCH_AMOUNT_MISMATCH`, `RECONCILIATION_LINE_INELIGIBLE`, `RECONCILIATION_NOT_BALANCED`, + period-lock 422). 10 `@EmitEvent` audit ids, all registered in `EventTypes`.
- **Permissions:** reuses the existing `accounting:reconciliation:view` (reads) / `accounting:reconciliation:adjust` (mutations) — **no CATALOG_VERSION bump**.
- **Why:** closes the D-5 reconciliation workstream — settlement reconciliation (F1c, Wave 4) plus this periodic bank-statement control on the cash accounts.

## Tests
- [x] Unit tests (`BankReconciliationServiceTest`, 15) — matching tolerance/N-to-1, finalize gate pass/fail, adjustment JE + period lock, not-reconcilable reject, CSV parse edge cases.
- [x] Controller MockMvc IT (`BankReconciliationControllerTest`, 19) — permission enforcement (view vs adjust) + error-code assertions.
- [x] ArchUnit (ADR-0011/0013) green.
- **How to run:** `./mvnw -pl pos-accounting -am test` — **full module suite 1092 green, 0 fail**.

## Risk & Rollback
- Risk level: Medium (new posting path + schema migration; repository access kept in the service layer; adjustment JEs are period-lock-gated and balanced).
- Rollback: revert the PR. V18 is additive/forward-only (drops only the unused `reconciliation` table).

## ⚠️ Accounting-owner sign-off requested (does not block backend correctness)
1. **Finalize identity.** The plan's formula `statement = glEndingBalance + Σ matched + Σ adjustments` double-counts: `glEndingBalance` is snapshotted **as-of the statement date**, so it already includes the matched GL lines (counterexample: bank 500 / GL 500 / one matched 500 deposit could never finalize). Implemented as `difference = statement − (glEndingBalance + Σ adjustments)`; matching is documentation (report shows matched-vs-outstanding), adjustments do the balancing. Alternative reading (keep Σ matched, snapshot the **opening** balance) is a one-line change — confirm intended identity.
2. **Advisory (deferred, as-specified):** per-adjustment-type sign guard (e.g. `BANK_FEE` must reduce cash) is not enforced; `FLOAT_ADJUSTMENT` posts a cash-moving JE (vs. tracking float as an outstanding item). Both are behavior decisions for the accounting owner.

## Checklist
- [x] Branch name matches `cap/<cap-id>`
- [x] PR title starts with `[CAP:<cap-id>]`
- [x] Links to parent + child issues present

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pos-accounting/src/main/java/com/positivity/accounting/internal/service/BankReconciliationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,579 @@
...
+        List<BankReconciliationGlMatch> matches = new ArrayList<>();
+        for (JournalEntryLine glLine : glLines) {
+            BankReconciliationGlMatch m = new BankReconciliationGlMatch();
+            m.setReconciliationId(reconciliationId);
+            m.setMatchId(matchId);
+            m.setGlLineId(glLine.getLineId());
+            m.setSignedAmount(signedGl(glLine));
+            matches.add(m);
+        }
+        glMatchRepository.saveAll(matches);
```
[the remark this answers is not in the document]
Comment (louisburroughs) [reply pos-accounting/src/main/java/com/positivity/accounting/internal/service/BankReconciliationServiceImpl.java]: Fixed in b489be7c1 — match now flushes the GL-match inserts inside a guard (saveAllAndFlush) and translates a concurrent unique(gl_line_id) DataIntegrityViolationException to ReconciliationLineIneligibleException → 409 RECONCILIATION_LINE_INELIGIBLE, not a 500. Covered by a new unit test.

File: pos-accounting/src/main/java/com/positivity/accounting/internal/service/BankReconciliationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+                if (line.getMatchId() != null) {
+                    matchIds.add(line.getMatchId());
+                }
+            }
+            if (matchIds.size() != 1) {
+                throw new IllegalArgumentException(
+                        "statementLineIds must resolve to exactly one match group; found " + matchIds.size());
+            }
+            matchId = matchIds.iterator().next();
+        }
```
[the remark this answers is not in the document]
Comment (louisburroughs) [reply pos-accounting/src/main/java/com/positivity/accounting/internal/service/BankReconciliationServiceImpl.java:306]: Fixed in b489be7c1 — unmatch(statementLineIds) now verifies every requested line exists and belongs to the target reconciliation before resolving the match group; unknown or cross-reconciliation ids return 404 RECONCILIATION_NOT_FOUND (was 400) and can no longer influence matchId resolution. Covered by a new unit test.

File: pos-accounting/src/main/java/com/positivity/accounting/internal/exception/ReconciliationNotBalancedException.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.positivity.accounting.internal.exception;
+
+import java.math.BigDecimal;
+
+/**
+ * Thrown when a bank reconciliation is finalized while it does not balance — the
+ * statement ending balance does not equal (GL ending balance + Σ matched + Σ
+ * adjustments) within ±0.01 (Story F2, issue #965). Maps to 422
+ * RECONCILIATION_NOT_BALANCED and surfaces the outstanding {@link #getDifference()
+ * difference}.
```
[the remark this answers is not in the document]
Comment (louisburroughs) [reply pos-accounting/src/main/java/com/positivity/accounting/internal/exception/ReconciliationNotBalancedException.java:10]: Fixed in c57f62188 — the ReconciliationNotBalancedException javadoc now reads '(GL ending balance as-of the statement date + Σ adjustments)', matching computeDifference(). The Σ matched term was the double-count corrected per the accounting-owner ruling on the finalize identity.
